### ISSUE TITLE
gmime-3.0 not found on Debian 10 (Buster)

### ISSUE THREAD

**Issue by thriveth** [djcb/mu#1679]
## Expected or desired behavior

Run `autogen.sh`, autogen is happy, tells me all is fine because I have run `sudo apt-get build-dep maildir-utils` right before this, and starts building. 

## Actual behavior

`./configure` or `autogen.sh` both complain that gmime-3.0 is not there, even though I have triple checked that libgmime-3.0 and associated -dev and -doc packages are installed.

Error message:

~~~
[...]
checking for GLIB... yes
checking for GMIME... no
configure: error: Package requirements (gmime-3.0) were not met:

No package 'gmime-3.0' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GMIME_CFLAGS
and GMIME_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
~~~

However, when I check for gmime-3 I get:

~~~
user@machine:/path/$  aptitude search gmime | grep 3
i A gir1.2-gmime-3.0 - MIME message parser and creator library - GObject introspection data
i A libgmime-3.0-0 - MIME message parser and creator library
i  libgmime-3.0-dev - MIME message parser and creator library - development files
i  libgmime-3.0-doc - MIME message parser and creator library - documentation
~~~

## Steps to reproduce

On my Debian 10 "Buster" system, navigated to the source dir, then:

~~~ bash
sudo apt update && sudo apt full-upgrade
sudo apt-get build-dep maildir-utils
./autogen.sh 
# ./configure if installing from source package, same problem
~~~

## Versions of mu, mu4e/emacs, operating system etc.

- Debian 10 "Buster".
- mu v. 1.4.4 *or*
- mu from GitHub repository

**Comment by djcb** [comment]
What happens when you do:
```
$ pkg-config --modversion gmime-3.0
```
?

**Comment by thriveth** [comment]
@djcb Thanks for your reply! 

The output I got was:

~~~
Package gmime-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gmime-3.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gmime-3.0' found
~~~

So I ran `locate gmime-3.0.pc`, added the relevant path to the `PKG_CONFIG_PATH` environment variable and ran `autogen.sh` again, and this time it worked.

However, when I run make, it now complains that it cannot find glib.h, although it is found in `/usr/include/glib-2.0`.

**Comment by djcb** [comment]
Hmmm, that you needed to update `PKG_CONFIG_PATH` suggests something is wrong since the default should work with the packages you installed with `apt`. The same is true for glib-2.0, i.e. 
```
$ pkg-config --modversion glib-2.0
```
should work, and you should get something like
```
$ pkg-config --cflags glib-2.0
-I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include 
```
(exact names may vary)

**Comment by thriveth** [comment]
Running `pkg-config --modversion glib-2.0` works and yields a version number (2.58.3).

Running `pkg-config --cflags glib-2.0` yields:

~~~
-I/usr/lib/include/glib-2.0 -I/usr/lib/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/lib/include
~~~

...None of which are where the actual `glib.h` file resides. 

I swear, I have not done anything funky with my paths and my Debian install.

**Comment by djcb** [comment]
There seems to be an extra `lib/` component there - very weird.

This is what I get on a Debian sid system:
```
% cat /etc/issue
Debian GNU/Linux bullseye/sid \n \l

$ pkg-config glib-2.0 --cflags
-I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include
```
perhaps some rogue environment variable?

**Comment by thriveth** [comment]
@djcb  Weird indeed. This is Buster, not Sid. Don't know if it has anything to do with it. 

Also, it's a Debian distro mirror maintained by the IT dept. at my university - there is a small possibility that they might have done something weird to it, although it would surprise me, they are usually very competent people. 

Any suggestions for a workaround?

**Comment by djcb** [comment]
Hopefully, we don't need a workaround!

Looking at
   https://packages.debian.org/buster/amd64/libgmime-3.0-dev/filelist
it seems gmime-3.0 should be in 
 `/usr/lib/x86_64-linux-gnu/pkgconfig/gmime-3.0.pc`
And I think the official Debian `pkg-config` (you didn't install some other version?)  should be able to find it there without any `PKG_*` environment variables (same for glib-2.0), if you have installed the dev-packages. 

What does `cat /usr/lib/x86_64-linux-gnu/pkgconfig/glib-2.0.pc` give you?

**Comment by thriveth** [comment]
@djcb  That is indeeedd where gmime-3.0.pc is located, but it only found it after I added that dir to the PKG_CONFIG_PATH. 

Here's the output of cat'ing glib-2.0

~~~
prefix=/usr
libdir=${prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include

bindir=${prefix}/bin
glib_genmarshal=${bindir}/glib-genmarshal
gobject_query=${bindir}/gobject-query
glib_mkenums=${bindir}/glib-mkenums

Name: GLib
Description: C Utility Library
Version: 2.58.3
Requires.private: libpcre >=  8.31
Libs: -L${libdir} -lglib-2.0
Libs.private: -pthread
Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include
~~~

**Comment by djcb** [comment]
That seems correct... so this declares that `/usr/include/glib-2.0` should be in your cflags, and hence `/usr/include/glib-2.0/glib.h` exists and will be found.

So where does it invent `-I/usr/lib/include/glib-2.0`? That's the problem here.

**Comment by thriveth** [comment]
Hmm, that is very strange.

**Comment by thriveth** [comment]
ARGH! Found the problem, at it was entirely my fault. 
I *had* been messing with my PATH - prepending Anaconda, which I did not in my wildest imagination think would have its own versions of glib... But it did. 

Also, it is strange that the anaconda path is not the one that showed up above. But removing that from the path made mu build.

So... make install ran smoothly. Now it tells me to reinit my database (built with 1.0, the version in Debian Stable)... Is this normal?
